### PR TITLE
Development retry deployment

### PR DESCRIPTION
**Pull request by m-motawea**
### Description

retry deployment on another node on failure in app chatflows

### CODE REVIEW

**Review by waleedhammam — APPROVED** [review]

**Review by ranatrk — APPROVED** [review]

File: jumpscale/sals/marketplace/apps_chatflow.py [hunk 1]
```diff
@@ -334,7 +338,20 @@ def get_solution_name(self):
...
+                success = True
+            except DeploymentFailed as e:
+                j.logger.error(e)
+                if self.retries > 0:
+                    self.md_show_update(f"Deployment failed on node {self.selected_node.node_id}. retrying....")
+                    self.retries -= 1
+                    self.ip_address = None
+                    self._deploy_network()
+                else:
+                    raise e
```
Comment (dmahmouali) [inline jumpscale/sals/marketplace/apps_chatflow.py:354]:
```suggestion
                    raise StopChatFlow(f"Failed to deploy this instance: { str(e) }")
```

File: jumpscale/sals/marketplace/apps_chatflow.py [hunk 2]
```diff
@@ -384,6 +401,30 @@ def solution_extension(self):
...
+                    )
+                    failed_workload = j.sals.zos.workloads.get(e.wid)
+                    if failed_workload.info.workload_type in GATEWAY_WORKLOAD_TYPES:
+                        self.addresses = []
+                        self._get_domain()
+                    else:
+                        self.ip_address = None
+                        self._deploy_network()
+                else:
+                    raise e
```
Comment (dmahmouali) [inline jumpscale/sals/marketplace/apps_chatflow.py:426]:
```suggestion
                    raise StopChatFlow(f"Failed to deploy this instance: { str(e) }")
```

**Review by AhmedSa-mir — APPROVED** [review]

File: jumpscale/sals/marketplace/apps_chatflow.py (hunk 1 above)
Comment (m-motawea) [reply jumpscale/sals/marketplace/apps_chatflow.py:354]: `e` is an instance of `DeploymentFailed`. I need to raise it exactly as it is to properly clean the workloads

File: jumpscale/sals/marketplace/apps_chatflow.py (hunk 2 above)
Comment (m-motawea) [reply jumpscale/sals/marketplace/apps_chatflow.py:426]: same as above comment

**Review by dmahmouali — APPROVED** [review]

**Pull request merged**